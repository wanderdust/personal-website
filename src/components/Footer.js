import React from 'react';

export const Footer = () => (
  <footer className="footer">
    <div>
      <ul className="footer__icons">
        <li>
          <a href="https://github.com/wanderdust?tab=repositories">
            <img className="footer__icon" src="/images/icon-github.svg" alt="Github" />
          </a>
        </li>
        <li>
          <a href="https://www.linkedin.com/in/lopezsantoripablo">
            <img className="footer__icon" src="/images/icon-linkedin.svg" alt="Linkedin" />
          </a>
        </li>
      </ul>
      <p>Icons from https://iconmonstr.com</p>
    </div>
  </footer>
);

export default Footer;
