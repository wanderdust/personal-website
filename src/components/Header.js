import React from 'react';
import { Link } from 'react-router-dom';

export const Header = () => (
  <header className="header">
    <div>
      <div className="header__content">

        <Link className="header-brand" to="/">
          <h2 className="header-brand__title">Pablo Lopez</h2>
          <h4 className="header-brand__subtitle">Front End Developer</h4>
        </Link>

        <div className="header__nav show-for-desktop">
          <div className="header__nav-item">
            <a className="button button--link" href="#profile"> Profile </a>
          </div>
          <div className="header__nav-item">
            <a className="button button--link" href="#experience"> Experience </a>
          </div>
          <div className="header__nav-item">
            <a className="button button--link" href="#abilities"> Abilities </a>
          </div>
          <div className="header__nav-item">
            <a className="button button--link" href="#projects"> Projects </a>
          </div>
          <div className="header__nav-item">
            <a className="button button--link" href="#contact"> Contact </a>
          </div>
        </div>
      </div>
    </div>
  </header>
);

export default Header;
