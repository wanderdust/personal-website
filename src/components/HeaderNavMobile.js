import React from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import uuid from 'uuid';
import { setIsNavbarOpen } from '../actions/mobileNav';

export const HeaderNavMobile = ({ isNavbarOpen, setNavbarStatus }) => {
  const headerLinks = ['Profile', 'Experience', 'Abilities', 'Projects', 'Contact'];

  const handleOnClick = () => {
    setNavbarStatus(!isNavbarOpen);
  };
  // For accesibilty for people without mouse.
  const handleOnKeyPress = (e) => {};

  const renderLinkButtons = () => headerLinks.map((linkName) => {
    const referralHash = `#${linkName.toLowerCase()}`;
    return (
      <div
        key={uuid()}
        onClick={handleOnClick}
        onKeyPress={handleOnKeyPress}
        role="presentation"
        className="header__nav-item mobile-header__nav-item"
      >
        <a className="button button--link" href={referralHash}>{linkName}</a>
      </div>
    );
  });

  return (
    <div className={`mobile-header show-for-mobile ${isNavbarOpen && 'mobile-header--open'}`}>
      {renderLinkButtons()}
    </div>
  );
};

HeaderNavMobile.propTypes = {
  isNavbarOpen: PropTypes.bool.isRequired,
  setNavbarStatus: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  isNavbarOpen: state.mobileNav.isNavbarOpen
});

const mapDispatchToProps = dispatch => ({
  setNavbarStatus: isNavbarOpen => dispatch(setIsNavbarOpen(isNavbarOpen))
});

export default connect(mapStateToProps, mapDispatchToProps)(HeaderNavMobile);
