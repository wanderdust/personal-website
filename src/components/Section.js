import React from 'react';
import propTypes from 'prop-types';

export const Section = ({ sectionTitle, sectionSubTitle, component: Component }) => (
  <div>
    <h1 className="section__title">{sectionTitle}</h1>
    <p className="section__subtitle">
      {sectionSubTitle}
    </p>
    <div className="section__content">
      <Component />
    </div>
  </div>
);

Section.propTypes = {
  sectionTitle: propTypes.string.isRequired,
  sectionSubTitle: propTypes.string.isRequired,
  component: propTypes.func.isRequired
};

export default Section;
