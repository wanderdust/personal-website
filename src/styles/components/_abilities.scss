.abilities__col {
  padding-top: 0;
  padding-bottom: 0;
}

.abilities__skills {
  display: flex;
  flex-direction: column;
}

.ability-item {
  display: flex;
  justify-content: space-between;
  align-items: center;
  &:hover {
    background: $light-gray;
    img {
      padding-left: 0.3rem;
    }
  }
}

.ability-item span {
  display: flex;
  flex-wrap: nowrap;
}

.ability-item__rating > img {
  height: 20px;
  width: auto;
  padding-left: 0;
  transition: padding .3s;
}


// Styles for desktop
@media (min-width: $desktop-breakpoint) {
  .abilities__col {
    padding: $m-size;
  }
  .abilities__skills {
    flex-direction: row;
  }
}