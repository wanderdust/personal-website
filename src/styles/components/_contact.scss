#contact {
  background-image: url("/images/bg.jpg");
  background-repeat: no-repeat;
  color: white;
  background-size: cover;
  background-color: black;
  opacity: 0.9;
}

.contact__content {
  display: flex;
  justify-content: center;
}

#contact .section__subtitle {
  margin-bottom: 2rem;
}