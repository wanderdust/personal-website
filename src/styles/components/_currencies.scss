.currencies__list {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
}

.currencies__column {
  display: flex;
  flex-direction: column;
  flex: 1;
}