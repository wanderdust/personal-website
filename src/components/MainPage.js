import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Section from './Section';
import Profile from './MainPageSections/Profile';
import Experience from './MainPageSections/Experience';
import Abilities from './MainPageSections/Abilities';
import Projects from './MainPageSections/Projects';
import Contact from './MainPageSections/Contact';

export const MainPage = ({
  profile, experience, abilities, projects
}) => (
  <div className="content-wrapper">
    <div id="profile" className="section section--basic">
      <Section
        sectionTitle={profile.title}
        sectionSubTitle={profile.profession}
        component={Profile}
      />
    </div>

    <div id="experience" className="section section--light-grey">
      <Section
        sectionTitle={experience.title}
        sectionSubTitle={experience.subtitle}
        component={Experience}
      />
    </div>

    <div id="abilities" className="section section--basic">
      <Section
        sectionTitle={abilities.title}
        sectionSubTitle={abilities.subtitle}
        component={Abilities}
      />
    </div>

    <div id="projects" className="section section--light-grey">
      <Section
        sectionTitle={projects.title}
        sectionSubTitle={projects.subtitle}
        component={Projects}
      />
    </div>

    <div id="contact" className="section section--basic">
      <Section
        sectionTitle="Get In Touch"
        // sectionSubTitle="Get in touch"
        component={Contact}
      />
      <div className="section-filter" />
    </div>
  </div>
);

MainPage.propTypes = {
  profile: PropTypes.object.isRequired,
  experience: PropTypes.object.isRequired,
  abilities: PropTypes.object.isRequired,
  projects: PropTypes.object.isRequired
};

const mapStateToProps = state => ({
  profile: state.data.profile,
  experience: state.data.experience,
  abilities: state.data.abilities,
  projects: state.data.projects
});

export default connect(mapStateToProps)(MainPage);
