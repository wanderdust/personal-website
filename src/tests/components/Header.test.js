/*
* shallow lets us test components using regular javascript.
* It only renders the given component.
*/
import React from 'react';
import { shallow } from 'enzyme';
import { Header } from '../../components/Header';

let isNavbarOpen;
let setNavbarStatus;
let wrapper;

beforeEach(() => {
  isNavbarOpen = true;
  setNavbarStatus = jest.fn();

  wrapper = shallow(
    <Header
      isNavbarOpen={isNavbarOpen}
      setNavbarStatus={setNavbarStatus}
    />
  );
});

test('should render header correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should handle onClick correctly', () => {
  wrapper.find('button').simulate('click');
  expect(setNavbarStatus).toHaveBeenCalledWith(false);
});
