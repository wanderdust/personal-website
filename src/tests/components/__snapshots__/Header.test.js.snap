// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header correctly 1`] = `
<header
  className="header"
>
  <div>
    <div
      className="header__content"
    >
      <Link
        className="header-brand"
        replace={false}
        to="/"
      >
        <h2
          className="header-brand__title"
        >
          Pablo Lopez
        </h2>
        <h4
          className="header-brand__subtitle"
        >
          Front End Developer
        </h4>
      </Link>
      <div
        className="header__nav show-for-desktop"
      >
        <div
          className="header__nav-item"
          key="1"
        >
          <a
            className="button button--link"
            href="#profile"
          >
            Profile
          </a>
        </div>
        <div
          className="header__nav-item"
          key="1"
        >
          <a
            className="button button--link"
            href="#experience"
          >
            Experience
          </a>
        </div>
        <div
          className="header__nav-item"
          key="1"
        >
          <a
            className="button button--link"
            href="#abilities"
          >
            Abilities
          </a>
        </div>
        <div
          className="header__nav-item"
          key="1"
        >
          <a
            className="button button--link"
            href="#projects"
          >
            Projects
          </a>
        </div>
        <div
          className="header__nav-item"
          key="1"
        >
          <a
            className="button button--link"
            href="#contact"
          >
            Contact
          </a>
        </div>
      </div>
      <button
        className="button button--link header__nav-item header__nav-item--mobile mobile-menu-trigger show-for-mobile"
        onClick={[Function]}
        onKeyPress={[Function]}
        type="button"
      >
        <span
          className="menu-icon-mobile"
        >
          <img
            alt="menu icon for mobile"
            src="images/icon-burger-menu.svg"
          />
        </span>
      </button>
    </div>
  </div>
</header>
`;
